readme.md

```diff
@@ -77,14 +315,190 @@ ___
+
+```typescript
+maxLength(customError?)
+```
+
+```typescript
+// Invalid value
+expect(email()('aninvalid.email')).toEqual('The email is not valid');
+// Valid value
+expect(email()('someone@somewhere.com')).toEqual(undefined);
+
+// Passing customError
+const errorMsg = 'Custom error.';
+expect(email(errorMsg)('aninvalid.email')).toEqual(errorMsg);
+```
+
 #### length
+The **length** validator validates the exact length of the textual data. The two parameters of this validator are : *size* and *customError*.
+| Name | Type | Description |
+| ----------- | ----------- | ----------- |
+| *size* | number | It represents the exact length of the textual data. |
+
+```typescript
+length(size, customError?)
+```
+
+```typescript
+// Invalid value
+const expectedError = `Should be 5 characters long`;
+expect(length(5)('1234')).toEqual(expectedError);expect(length(5)('1234')).toEqual(expectedError);
+// Valid value
+expect(length(1)('1')).toEqual(undefined);
+
+// Passing customError
+const errorMsg = 'Custom error.';
+expect(length(5, errorMsg)('1')).toEqual(errorMsg);
+```
+
 #### pattern
+The **pattern** validator checks if a value matches a specific pattern. It takes two parameters: *regex* and *customError*.
+| Name | Type | Description |
+| ----------- | ----------- | ----------- |
+| *regex* | regular expression | A pattern that entered data needs to follow. |
+```typescript
+pattern(regex, customError?)
+```
+
+```typescript
+// Invalid value
+expect(pattern(/^[a-b]/)('z')).toEqual(`The value 'z' doen't match the pattern /^[a-b]/`);
+// Valid value
+expect(pattern(/^[a-z0-9]*$/)('12ba')).toEqual(undefined);
+
+// Passing customError
+const errorMsg = 'Custom error.';
+expect(pattern(/^[a-z0-9]*$/, errorMsg)('abc_123')).toEqual(errorMsg);
+```
+
 #### min
+The **min** validator checks whether a value is greater than the minimum value specified. The two parameters of the validators are: *number* and *customError*.
+| Name | Type | Description |
+| ----------- | ----------- | ----------- |
+| *number* | number | Minimum value that entered data needs to be greater than. |
+```typescript
+min(number, customError?)
+```
+
+```typescript
+// Invalid value
+expect(min(15)(12)).toEqual(`Value 12 should be greater than 15`);
+// Valid value
+expect(min(15)(17)).toEqual(undefined);
+
+// Passing customError
+const errorMsg = 'Custom error.';
+expect(min(6, errorMsg)(5)).toEqual(errorMsg);
+```
+
 #### max
+The **max** validator checks whether a value is smaller than the maximum value specified. The two parameters of the validators are: *number* and *customError*.
+| Name | Type | Description |
+| ----------- | ----------- | ----------- |
+| *number* | number | Maximum value that entered data needs to be smaller than. |
+
+```typescript
+max(number, customError?)
+```
+
+```typescript
+// Invalid value
+expect(max(12)(15)).toEqual(`Value 15 should be smaller than 12`);
```

Comment:
Can we use validators along with respective decoders?

Reply:
Like using min and max validator after number decoder?

Reply:
Yep. Reason being, validators will always be used with decoders.